lib/linguist/heuristics.yml

```
@@ -277,6 +277,13 @@ disambiguations:
   - language: NL
     pattern: '^(b|g)[0-9]+ '
   - language: NewLisp
+- extensions: ['.p']
+  rules:
+  - language: Gnuplot
+    pattern:
+    - '^s?plot\b'
+    - '^set\s+(term|terminal|out|output|[xy]tics|[xy]label|[xy]range|style)\b'
+  - language: OpenEdge ABL
```

It would be better if we could find a pattern for OpenEdge ABL as well.

/cc @abevoelker, who added support for OpenEdge ABL in Linguist (#115), in case they have an idea.

---

It's been quite a few years but I think a pattern like this should catch most Progress / OpenEdge ABL programs:

```suggestion
  - language: OpenEdge ABL
    pattern: '^\s*(?i:(DEFINE|FOR|PROCEDURE|MESSAGE|RUN|ASSIGN|REPEAT|CLASS|METHOD|&SCOPED-DEFINE|&MESSAGE|&GLOBAL-DEFINE))'
```